### PR TITLE
Add support to set PSU fan speed

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
**- What I did**

1. Add support to set PSU fan speed via i2cset tool
2. Add a private policy to update PSU fan speed timely

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/tests/test_thermal_policy.py [hunk 1]
```diff
@@ -499,7 +499,7 @@ def test_dynamic_minimum_policy(thermal_manager):
 
     Thermal.get_air_flow_direction = MagicMock(return_value=('c2p', 25000))
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/tests/test_thermal_policy.py:500]: why here the temperature not changed from "25000" to "25"?

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py [hunk 2]
```diff
@@ -52,6 +53,21 @@ def execute(self, thermal_info_dict):
...
+    @classmethod
+    def set_psu_fan_speed(cls, thermal_info_dict, speed):
+        from .thermal_infos import ChassisInfo
+        if ChassisInfo.INFO_NAME in thermal_info_dict and isinstance(thermal_info_dict[ChassisInfo.INFO_NAME], ChassisInfo):
+            chassis = thermal_info_dict[ChassisInfo.INFO_NAME].get_chassis()
+            for psu in chassis.get_all_psus():
+                for psu_fan in psu.get_all_fans():
+                    psu_fan.set_speed(speed)
+
+            logger.log_info('Updated PSU FAN speed to {}%'.format(speed))
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py:69]: is the fan speed here percentage or RPM? if PRM shouldn't have a '%'?

File: platform/mellanox/mlnx-platform-api/tests/test_thermal_policy.py (hunk 1 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/tests/test_thermal_policy.py:500]: it wiil divide by 1000 in the condition.is_match

File: platform/mellanox/mlnx-platform-api/sonic_platform/fan.py [hunk 3]
```diff
@@ -239,9 +248,19 @@ def set_speed(self, speed):
...
-        
+            try:
+                with open(self.psu_i2c_bus_path, 'r') as f:
+                    bus = f.read().strip()
+                with open(self.psu_i2c_addr_path, 'r') as f:
+                    addr = f.read().strip()
+                with open(self.psu_i2c_command_path, 'r') as f:
+                    command = f.read().strip()
+                speed = Fan.PSU_FAN_SPEED[int(speed / 10)]
+                subprocess.call("i2cset -f -y {0} {1} {2} {3} wp".format(bus, addr, command, speed), shell = True)
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/fan.py:259]: do we need to consider the failure case of executing this command?

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py (hunk 2 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py:69]: It is percentage.

File: platform/mellanox/mlnx-platform-api/sonic_platform/fan.py (hunk 3 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/fan.py:259]: IMO, we can catch exception and retry. What's your suggestion?

Comment (keboliu) [reply platform/mellanox/mlnx-platform-api/sonic_platform/fan.py:259]: use subprocess.check_call and check the return value?

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py (hunk 2 above)
Comment (keboliu) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py:69]: I thought we don't konw the max speed of PSU fan?

Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal_actions.py:69]: we don't know the max speed, but we have a vector which contains 11 values, so if cooling level is 5, we use the 6th element to set the PSU FAN speed.

File: platform/mellanox/mlnx-platform-api/sonic_platform/fan.py (hunk 3 above)
Comment (Junchao-Mellanox) [reply platform/mellanox/mlnx-platform-api/sonic_platform/fan.py:259]: sure

**Comment by Junchao-Mellanox** [comment]
Community PR https://github.com/Azure/sonic-buildimage/pull/4403

**Pull request closed without merging**